File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/behavior/BehaviorAppBarCollapse.java
package com.sup.dev.android.views.behavior;

import android.content.Context;
import android.support.design.widget.CoordinatorLayout;
import android.util.AttributeSet;
import android.view.View;

public class BehaviorAppBarCollapse<V extends View> extends BehaviorCollapse<V> {

    public BehaviorAppBarCollapse(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onDependentViewChanged(CoordinatorLayout parent, V child, View dependency) {
        child.setY(dependency.getY() + dependency.getHeight() - child.getHeight() / 2);
        return super.onDependentViewChanged(parent, child, dependency);
    }
}

File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/behavior/BehaviorAppBarCollapse.kt
package com.sup.dev.android.views.behavior

import android.content.Context
import android.support.design.widget.CoordinatorLayout
import android.util.AttributeSet
import android.view.View


class BehaviorAppBarCollapse<V : View>(context: Context, attrs: AttributeSet) : BehaviorCollapse<V>(context, attrs) {

    override fun onDependentViewChanged(parent: CoordinatorLayout?, child: V?, dependency: View?): Boolean {
        child!!.y = dependency!!.y + dependency.height - child.height / 2
        return super.onDependentViewChanged(parent, child, dependency)
    }
}